Tests/Realm.Tests/Database/RealmObjectTests.cs

```diff
@@ -122,6 +123,24 @@ public void RealmObject_AddToRelatedList_InvokesOnManaged()
             Assert.That(first.RelatedCollection[0].OnManagedCalled, Is.EqualTo(1));
         }
 
+        [Test]
+        public void RealmObject_ObjectSchema_ReturnsValueWhenManaged()
```

Comment:
I noticed we weren't covering both branches of `RealmObject.ObjectSchema`, so added this test.